libsolidity/formal/CHC.cpp

```diff
@@ -61,11 +61,62 @@ bool CHC::visit(ContractDefinition const& _contract)
 	if (!shouldVisit(_contract))
 		return false;
 
+	m_contract = &_contract;
+
 	reset();
 
 	if (!SMTEncoder::visit(_contract))
 		return false;
 
+	for (auto const& contract: _contract.annotation().linearizedBaseContracts)
```

Comment:
Maybe pull this block into a seperate function `readStateVariables` ?

Reply:
Makes sense